plugins/json_importer.cc

```diff
@@ -251,22 +322,40 @@ struct JSONImporter: public Procedure {
                 return handleError(exc.what(), actualLineNum, string(line, lineLength));
             }
 
-            if (useWhere) {
-                JsonRowScope row(expr);
-                if (!whereBound(row, storage, GET_ALL).isTrue()) {
-                    return true;
-                }
-            }
-
             skipJsonWhitespace(*parser.context);
             if (!parser.context->eof()) {
                 return handleError("extra characters at end of line", actualLineNum, "");
             }
 
+            unique_ptr<RowName> rowName;
```

Comment:
Don't use a unique_ptr here.  It adds a useless alloc.  Use a rowName and assign it directly.